File: app/src/main/java/khorunzhyicom/alex/theposter/presentation/top/TopMoviesController.java
package khorunzhyicom.alex.theposter.presentation.top;

import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.hannesdorfmann.mosby3.mvp.conductor.MvpController;

import khorunzhyicom.alex.theposter.R;
import khorunzhyicom.alex.theposter.application.App;
import khorunzhyicom.alex.theposter.di.components.injector.PresentationInjector;
import khorunzhyicom.alex.theposter.presentation.common.controller.RxMvpController;

public class TopMoviesController extends RxMvpController<TopMoviesView, TopMoviesPresenter> implements TopMoviesView {

    @NonNull
    @Override
    protected View onCreateView(@NonNull LayoutInflater layoutInflater, @NonNull ViewGroup viewGroup) {
        return layoutInflater.inflate(R.layout.controller_top_movies, viewGroup, false);
    }

    @NonNull
    @Override
    public TopMoviesPresenter createPresenter() {
        return new TopMoviesPresenter();
    }

    @Override
    public PresentationInjector injector() {
        return ((App)getApplicationContext()).apiComponent();
    }
}

File: app/src/main/java/khorunzhyicom/alex/theposter/presentation/top/TopMoviesController.kt
package khorunzhyicom.alex.theposter.presentation.top

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import khorunzhyicom.alex.theposter.R
import khorunzhyicom.alex.theposter.presentation.common.controller.PosterTabController
import khorunzhyicom.alex.theposter.presentation.common.controller.binder.ViewBinderController

class TopMoviesController(args: Bundle? = null) : PosterTabController<TopMoviesView, TopMoviesPresenter>(args), TopMoviesView {

    override fun inflateView(inflater: LayoutInflater, container: ViewGroup): View {
        return inflater.inflate(R.layout.controller_top_movies, container, false)
    }

    override fun createPresenter(): TopMoviesPresenter =  TopMoviesPresenter()

    override fun tabTitle(): Int = R.string.tab_top

    override fun tabController(): TopMoviesController = this

}
